flink-runtime/src/main/java/org/apache/flink/runtime/state/ttl/TtlValueState.java

```diff
@@ -48,7 +48,11 @@ public T value() throws IOException {
     @Override
     public void update(T value) throws IOException {
         accessCallback.run();
-        original.update(wrapWithTs(value));
+        if (value == null) {
```

Comment:
How about just make it like: `original.update(value == null ? null : wrapWithTs(value));` ?
Just let the original one to process the logic of updating null which should also follow the contract.